### PR TITLE
Revert TensorFlow/Keras/Python upgrade to retain support for M1 Macs

### PR DESCRIPTION
**Pull request by joshuacwnewton**
#### GitHub

- [ ] I've given this PR a concise, self-descriptive, and meaningful title
- [ ] I've linked relevant issues in the PR body
- [ ] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [ ] I've applied a [release milestone](https://github.com/spinalcordtoolbox/spinalcordtoolbox/milestones) (major, minor, patch) in line with [Semantic Versioning guidelines](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release#convention-for-naming-releases) 
- [ ] I've assigned a reviewer

#### PR contents

- [ ] I've consulted [SCT's internal developer documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

This PR reverts:

* #3361
* #3561

This brings us back to TensorFlow 1.5, which restores compatibility with AVX-less CPUs. 

However, this will also re-open #3227, so this shouldn't be considered a long-term solution. Ideally, we would find a way to upgrade Tensorflow/Python _and_ retain support for M1 macs.

## Linked issues

This is a temporary solution to #3566, so I believe that issue should stay open even if this is merged.

### CODE REVIEW

**Comment by joshuacwnewton** [comment]
I'm going to close this now, because:

* Short-term, 5.4 is not affected by this issue.
* Long-term, I think https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3569 will be a better solution.
